### DISCUSSION TITLE
[CRUD Service] GET /count with error in _q returns 500 instead of 400

### DISCUSSION THREAD

**Discussion by GioAc96** [mia-platform/community#86] [Q&A] [answered]
When querying the `/count` endpoint on the CRUD service with a bad value set to the `_q` parameter, the CRUD returns `500` instead of `400`.

Here's the URL I'm using: `http://crud-service/pending-passes/count_q=%7B%22tariff%22%3A13100%7D&_st=PUBLIC%2CDRAFT`
Here's the response I'm getting:
```
{"statusCode":500,"error":"Internal Server Error","message":"Unknown field: tariff"}
```

Is this the intended behaviour, or should the CRUD service respond with a different status code?

**Comment by fredmaggiowski** [comment, marked as answer]
The CRUD service tries to parse the query and currently throws an error which is interpreted as a 500; i agree that it could be better specified as a Bad Request (400).

**Reply by GioAc96** [reply]
Should I open an issue on the CRUD service repository and mark this discussion as answered?

**Reply by fredmaggiowski** [reply]
I'd do both things, I think the question in itself is answered; regardless of this I'd open an issue so that this issue can be actually addressed

**Reply by GioAc96** [reply]
Here it is: https://github.com/mia-platform/crud-service/issues/55

**Reply by fredmaggiowski** [reply]
@danibix95 @TheRusCompass FYI

**Reply by danibix95** [reply]
Hi, thanks for the reminder and for creating the issue, so that we can keep track of it.
